c7n/resources/iam.py

```diff
@@ -440,6 +440,88 @@ class resource_type(TypeInfo):
         global_resource = True
 
 
+@ServerCertificate.action_registry.register('delete')
+class CertificateDelete(BaseAction):
+    """Delete an IAM Certificate
+
+    For example, if you want to automatically delete an unused IAM certificate.
+
+    :example:
+
+      .. code-block:: yaml
+
+        - name: aws-iam-certificate-delete-expired
+          resource: iam-certificate
+          filters:
+            - type: value
+              key: Expiration
+              value_type: expiration
+              op: greater-than
+              value: 0
+          actions:
+            - type: delete
+
+    """
+    schema = type_schema('delete')
+    permissions = ('iam:DeleteServerCertificate',)
+
+    def validate(self):
+        # TODO: Let's argue
+        pass
+
+    def process(self, resources):
+
+        # get reference to AWS client
+        client = local_session(self.manager.session_factory).client('iam')
+
+        # for each resource call delete
+        for cert in resources:
+            self.manager.retry(
+                client.delete_server_certificate,
+                ServerCertificateName=cert['ServerCertificateName'],
```

Comment:
pass ignore_err_code to not raise for certificates already deleted/in progress on deletion.